userprofile/static/userprofile/extra_logins.js

```diff
@@ -2,16 +2,17 @@
 
   function showExtras(event) {
     event.preventDefault();
-    const extras = $("#login-box-row").find(".extra-login")
+    const extras = $("#login-box-row").find(".extra-login");
     extras.show();
     $(".show-extra-login-btn").hide();
     extras.find(":input:visible").first().focus() // focus for keyboard navigation
   }
 
   $(function () {
-    const row = $("#login-box-row")
-    const normal = row.find(".login-box").not(".extra-login")
-    if (normal.length > 0) {
+    const row = $("#login-box-row");
+    const normal = row.find(".login-box").not(".extra-login");
+    const errors = row.find(".extra-login").find(".alert");
+    if (normal.length > 0 && errors.length == 0) {
```

Comment:
This is fine, but maybe there should be a comment to mention that this now doesn't hide any of the extra logins if any of them have errors. Alternatively, I think you could do `row.find('.extra-login').filter(function(b) { return !b.find('.alert'); }).hide();`.

Reply:
I think it's better if none of the extra logins are hidden, so the view stays the same (so if there were several, logins stay in the same spot and none disappear). Otherwise that could have been implemented in the template with the extra-login class being conditional. But yeah, I'll comment that!

Reply:
A good argument!